DedicatedServer/Scripts/Libs/Zrx/ModeLibs/Common/Collision/CylinderHitbox.Script.txt

```diff
@@ -160,65 +164,34 @@ Boolean AreTouching(CSmPlayer _Player1, CSmPlayer _Player2) {
-	declare Real Player2Pitch = Private_GetPitch(_Player2.AimPitch);
-	
-	// ---------------------------------- //
-	// Cylinder diameter depends on player rotation
-	declare Real Player1Diameter;
-	Player1Diameter = (C_CarModelSizes[MapPlayerModelName].X - C_VerticalSizeSub) * Private_GetCyliDiameter(Player1Pitch);
-	declare Real Player2Diameter;
-	Player2Diameter = (C_CarModelSizes[MapPlayerModelName].X - C_VerticalSizeSub) * Private_GetCyliDiameter(Player2Pitch);
-	
-	// ---------------------------------- //
-	// Diameter can't be shorter than vehicle width
-	if (Player1Diameter < C_CarModelSizes[MapPlayerModelName].Z) Player1Diameter = C_CarModelSizes[MapPlayerModelName].Z;
-	if (Player2Diameter < C_CarModelSizes[MapPlayerModelName].Z) Player2Diameter = C_CarModelSizes[MapPlayerModelName].Z;
-	
-	// ---------------------------------- //
-	// Check if cylinders touch in X and Z axis
-	if (Private_Distance2D(FixedPos1, FixedPos2) > (Player1Diameter + Player2Diameter) / 2) return False;
-	
-	// ---------------------------------- //
-	// Cylinder height depends on player rotation
-	declare Real Player1Height = C_CarModelSizes[MapPlayerModelName].X * Private_GetCyliHeight(Player1Pitch);
-	declare Real Player2Height = C_CarModelSizes[MapPlayerModelName].X * Private_GetCyliHeight(Player2Pitch);
-	
-	// ---------------------------------- //
-	// Height can't be lower then vehicle height
-	if (Player1Height < C_CarModelSizes[MapPlayerModelName].Y) Player1Height = C_CarModelSizes[MapPlayerModelName].Y;
-	if (Player2Height < C_CarModelSizes[MapPlayerModelName].Y) Player2Height = C_CarModelSizes[MapPlayerModelName].Y;
-	
-	// ---------------------------------- //
-	// Check if cylinders touch in Y axis
-	if (Private_Distance1D(FixedPos1.Y, FixedPos2.Y) - (C_VerticalSizeSub * 2) > (Player1Height + Player2Height) / 2) return False;
-	
-	// ---------------------------------- //
-	// Players touch
-	return True;
+	declare Vec3 FixedPos1Front = _Player1.Position + _Player1.AimDirection * 1.184;
+	declare Vec3 FixedPos2Front = _Player2.Position + _Player2.AimDirection * 1.184;	
+	declare Vec3 FixedPos1Back = _Player1.Position - _Player1.AimDirection * 0.553;
+	declare Vec3 FixedPos2Back = _Player2.Position - _Player2.AimDirection * 0.553;
+
+	declare Real Multiplier = 1.;
+	if (UsePvPCollisions) Multiplier = 1.34;
+	if (Private_Distance2D(FixedPos1Front, FixedPos2Front) < 2. * Multiplier && MathLib::Abs(FixedPos1Front.Y - FixedPos2Front.Y) < 1. * Multiplier) return True;
+	if (Private_Distance2D(FixedPos1Front, FixedPos2Back) < 2.07 * Multiplier && MathLib::Abs(FixedPos1Front.Y - FixedPos2Back.Y) < 1. * Multiplier) return True;
+	if (Private_Distance2D(FixedPos1Back, FixedPos2Front) < 2.07 * Multiplier && MathLib::Abs(FixedPos1Back.Y - FixedPos2Front.Y) < 1. * Multiplier) return True;
+	if (Private_Distance2D(FixedPos1Back, FixedPos2Back) < 2.14 * Multiplier && MathLib::Abs(FixedPos1Back.Y - FixedPos2Back.Y) < 1. * Multiplier) return True;
```

Comment:
The pitch of the car is never taken into account. How does it work, if the player is pitched? This might work when both cars a flat on the ground, but otherwise it might be very unprecise.
Also, what do these constant values represent?